gammapy/data/event_list.py

```diff
@@ -382,22 +384,37 @@ def select_energy(self, energy_range):
         mask &= energy < energy_range[1]
         return self.select_row_subset(mask)
 
-    def select_time(self, time_interval):
-        """Select events in time interval.
+    def select_time(self, time_intervals):
+        """Select events in time intervals.
 
         Parameters
         ----------
-        time_interval : `astropy.time.Time`
+        time_intervals : `astropy.time.Time` or list of time_interval
             Start time (inclusive) and stop time (exclusive) for the selection.
 
         Returns
         -------
         events : `EventList`
             Copy of event list with selection applied.
+
+        Notes
+        -----
+        This function does not modify the metadata, in particular the ONTIME. It is instead recommended to use an
+        `~gammapy.data.ObservationFilter` associated to the object `~gammapy.data.Observation`.
         """
+        if time_intervals is None:
+            return self
+
         time = self.time
-        mask = time_interval[0] <= time
-        mask &= time < time_interval[1]
+        mask = np.full(len(time), False)
+        if time_intervals.value.shape != (2,):
+            for time_interval in time_intervals:
+                submask = time_interval[0] <= time
+                submask &= time < time_interval[1]
+                mask |= submask
```

Comment:
The mask would get written to the latest one.

Reply:
I did not get your point... This is an OR made for each interval and it is updated at each loop